docs/by-example/sqlalchemy/advanced-querying.rst

```diff
@@ -0,0 +1,267 @@
+
+    >>> session = sessionmaker(bind=engine)()
+    >>> Base = declarative_base()
+
+
+Introduction to fulltext indexes
+================================
+
+:ref:`crate-reference:fulltext-indices` take the contents of one or more fields
+and split it up into tokens that are used for fulltext-search. The transformation
+from a text to separate tokens is done by an analyzer. In order to create
+fulltext search queries a :ref:`fulltext index with an analyzer
+<crate-reference:sql_ddl_index_fulltext>` must be defined for the related columns.
+
+In order to support fulltext query operations, the CrateDB SQLAlchemy dialect
+provides the :ref:`crate-reference:predicates_match` through its ``match`` function.
+
+For exercising those features, let's define a schema using SQLAlchemy's
+:ref:`sa:orm_declarative_mapping`:
+
+    >>> def gen_key():
+    ...     return str(uuid4())
+
+    >>> class Character(Base):
+    ...     __tablename__ = 'characters'
+    ...     id = sa.Column(sa.String, primary_key=True, default=gen_key)
+    ...     name = sa.Column(sa.String)
+    ...     quote = sa.Column(sa.String)
+    ...     name_ft = sa.Column(sa.String)
+    ...     quote_ft = sa.Column(sa.String)
+    ...     __mapper_args__ = {
+    ...         'exclude_properties': ['name_ft', 'quote_ft']
+    ...     }
+
+Please note that the schema defined above is provisioned to the database using
+the following SQL DDL statement.
+
+.. code-block:: sql
+
+    CREATE TABLE characters (
+        id STRING PRIMARY KEY,
+        name STRING,
+        quote STRING,
+        INDEX name_ft USING fulltext(name) WITH (analyzer = 'english'),
+        INDEX quote_ft USING fulltext(quote) WITH (analyzer = 'english')
+    )
+
+.. note::
+
+    Currently, it is not supported to define a *named index column definition*
+    using a :ref:`crate-reference:sql_ddl_index_fulltext` on behalf of the
+    SQLAlchemy declarative schema.
+
+Let's add two records we use for testing.
+
+    >>> arthur = Character(name='Arthur Dent')
+    >>> arthur.quote = "Let's go somewhere."
+    >>> session.add(arthur)
+
+    >>> trillian = Character(name='Tricia McMillan')
+    >>> trillian.quote = "We're on a space ship Arthur. In space."
+    >>> session.add(trillian)
+
+    >>> session.commit()
+
+After ``INSERT`` statements are submitted to the database, the newly inserted
+records aren't immediately available for retrieval because the index is only
+updated periodically (default: each second). In order to synchronize that,
+refresh the table:
+
+    >>> _ = connection.execute(sa.text("REFRESH TABLE characters"))
+
+
+Aggregates: Counting and grouping
+=================================
+
+SQLAlchemy supports different approaches to issue a query with a count
+aggregate function. Take a look at the `count result rows`_ documentation
+for a full overview.
+
+CrateDB currently does not support all variants as it can not handle the
+sub-queries yet.
```

Comment:
Is this still the case?

Reply:
Good question. I also wanted to highlight this note while reading it, but missed to do so. Which suggestion do you have?
a) Investigate the matter.
b) Just remove this sentence.

Reply:
I will do. However, to move on with this, and other subsequent patches, I've deferred it to GH-489. I hope this is fine.